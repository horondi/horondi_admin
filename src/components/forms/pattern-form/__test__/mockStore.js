const mockStore = {
  materialsByPurpose: [
    {
      _id: 1,
      name: [
        {
          value: 'Backpack'
        }
      ]
    }
  ],
  loading: false
};

export default mockStore;
