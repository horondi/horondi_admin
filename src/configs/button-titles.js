const buttonTitles = {
  DELETE_TITLE: 'Видалити',
  EDIT_TITLE: 'Редагувати',
  CREATE_NEWS_TITLE: 'Додати новину',
  CREATE_PATTERN_TITLE: 'Додати гобелен',
  CREATE_MODEL_TITLE: 'Додати модель',
  MODEL_SAVE_TITLE: 'Зберегти',
  CANCEL_TITLE: 'Відмінити',
  LOGOUT_TITLE: 'Вихід',
  CREATE_BUSINESS_PAGE: 'Додати бізнес сторінку',
  CREATE_CONTACT_TITLE: 'Додати контакти',
  SWITCH_USER_STATUS_TITLE: 'Змінити статус користувача',
  USER_INACTIVE_TITLE: 'Деактивувати',
  USER_ACTIVE_TITLE: 'Активувати',
  ADD_CATEGORY: 'Додати категорію',
  GO_BACK_TITLE: 'Назад',
  ADD_SUBCATEGORY: 'Додати підкатегорію',
  ADD_CATEGORY_IMAGE: 'Додати зображення',
  ADD_CATEGORY_NAME: 'Додати назву',
  CANCEL: 'Відмінити',
  SAVE_CATEGORY: 'Зберегти категорію',
  SAVE_SUBCATEGORY: 'Зберегти підкатегорію',
  CREATE_SPECIAL_USER: 'Створити спецкористувача',
  ADD_PHOTO_LABEL: '+',
  CREATE_CATEGORY: 'Створити категорію',
  CREATE_SUBCATEGORY: 'Створити підкатегорію',
  PATTERN_REMOVE_MESSAGE: 'Ви впевнені, що хочете видалити цей гобелен?',
  REMOVE_CONTACT_MESSAGE: 'Ви впевнені, що хочете видалити цей контакт?',
  USER_UNACTIVE_TITLE: 'Деактивувати',
  SHOW_COMMENTS_TITLE: 'Переглянути коментарі',
  HIDE_COMMENTS_TITLE: 'Приховати коментарі',
  SAVE_ORDER: 'Зберегти замовлення',
  SAVE_TITLE: 'Зберегти',
  UNDERSTAND: 'Зрозуміло',
  ANSWER: 'Відповісти',
  TO_SPAM: 'У СПАМ',
  MOVE_ALL_TO_SPAM: 'Перемістити у СПАМ',
  CREATE_MATERIAL_TITLE: 'Створити матеріал',
  CREATE_COLOR_TITLE: 'Створити колір',
  SAVE_MATERIAL: 'Зберегти матеріал',
  CREATE_HEADER_TITLE: 'Створити посилання',
  HEADER_SAVE_TITLE: 'Зберегти посилання',
  GO_TO_MATERIAL_COLOR_PALLET: 'Редагувати палітру',
  CLOSE_DIALOG_TITLE: 'Закрити вікно',
  ORDER_DETAILS: 'Деталі',
  USER_STATUS_TITLE: 'Статус',
  CLEAR_FILTERS: 'Очистити фільтри',
  CREATE_SLIDE_TITLE: 'Створити слайд',
  OPEN_SLIDE_EDIT: 'Увімкнути редагування',
  SAVE_SLIDE_ORDER: 'Зберегти зміни',
  CANCEL_SLIDE_ORDER: 'Скасувати зміни',

  titleGenerator: (editMode, isMain) => {
    const editModeMap = new Map([
      [true, 'Зберегти'],
      [false, 'Створити']
    ]);
    const isMainMap = new Map([
      [true, 'категорію'],
      [false, 'підкатегорію']
    ]);
    return `${editModeMap.get(editMode)} ${isMainMap.get(isMain)}`;
  }
};
export default buttonTitles;
