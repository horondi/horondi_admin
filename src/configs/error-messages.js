export const errorMessages = {
  USER_NOT_FOUND: 'Користувач не знайдений!',
  USER_NOT_AUTHORIZED: 'Користувач не отримав прав доступу',
  INVALID_PERMISSIONS: 'Недостатньо прав користувача',
  WRONG_CREDENTIALS: 'Неправильно вказані вхідні дані',
  INPUT_NOT_VALID: 'Неправильні ввідні дані',
  USER_ALREADY_EXIST: 'Користувач з такими даними вже існує',
  INVALID_ADMIN_INVITATIONAL_TOKEN:
    'Неправильне посилання на створення користувача',
  NO_COLORS: 'Створіть колір'
};

export const loginErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PASSWORD_MIN_LENGTH_MESSAGE: 'Пароль повинен містити не менше 8 символів',
  PASSWORD_MAX_LENGTH_MESSAGE: 'Пароль повинен містити не більше 20 символів',
  PASSWORD_LANG_MESSAGE: 'Використовуйте латиницю різних регістрів та цифри',
  ENTER_FIRSTNAME_MESSAGE: "Введіть ім'я",
  ENTER_LASTNAME_MESSAGE: 'Введіть прізвище',
  ENTER_PASSWORD_MESSAGE: 'Введіть пароль',
  FIRSTNAME_MIN_LENGTH_MESSAGE: "Ім'я повинно містити не менше 2 символів",
  LASTNAME_MIN_LENGTH_MESSAGE: 'Прізвище повинно містити не менше 2 символів',
  FIRSTNAME_MAX_LENGTH_MESSAGE: "Ім'я повинно містити не більше 30 символів",
  LASTNAME_MAX_LENGTH_MESSAGE: 'Прізвище повинно містити не більше 30 символів',
  SELECT_ROLE_MESSAGE: 'Оберіть роль'
};
export const patternErrorMessages = {
  PATTERN_VALIDATION_ERROR: 'Мінімум 2 символи',
  PATTERN_ERROR_MESSAGE: 'Поле не може бути порожнім',
  PATTERN_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри',
  PHOTO_NOT_PROVIDED: "Фото гобелена є обов'язковим",
  CONSTRUCTOR_PHOTO_NOT_PROVIDED: "Фото для конструктора є обов'язковим",
  PATTERN_UA_NAME_MESSAGE:
    'Поле може містити тільки українські літери та цифри',
  PATTERN_EN_NAME_MESSAGE: 'Поле може містити тільки англійські літери та цифри'
};
export const categoryErrorMessages = {
  CATEGORY_VALIDATION_ERROR: 'Мінімум 2 символи',
  CATEGORY_ERROR_MESSAGE: 'Поле не може бути порожнім',
  CATEGORY_UA_NAME_MESSAGE:
    'Поле може містити тільки українські літери та цифри',
  CATEGORY_EN_NAME_MESSAGE:
    'Поле може містити тільки англійські літери та цифри',
  CATEGORY_CODE_MESSAGE: 'Поле не може містити спеціальні символи'
};
export const headerErrorMessages = {
  HEADER_VALIDATION_ERROR: 'Мінімум 2 символи',
  HEADER_ERROR_MESSAGE: 'Поле не може бути порожнім',
  HEADER_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри'
};
export const newsErrorMessages = {
  NAME_MIN_LENGTH_MESSAGE: `Ім'я автора повинне містити не менше 2 символів`,
  TITLE_MIN_LENGTH_MESSAGE: 'Заголовок повинен містити не менше 10 символів',
  TEXT_MIN_LENGTH_MESSAGE: 'Текстове поле повино містити не менше 10 символів'
};
export const modelErrorMessages = {
  MODEL_VALIDATION_ERROR: 'Мінімум 2 символи',
  MODEL_ERROR_MESSAGE: 'Поле не може бути порожнім',
  MODEL_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри',
  PHOTO_NOT_PROVIDED: 'Додайте фото для моделі'
};
export const contactErrorMessages = {
  INVALID_EMAIL_MESSAGE: 'Некоректна email адреса',
  ENTER_EMAIL_MESSAGE: 'Введіть email',
  PHONE_NUMBER_LENGTH_MESSAGE:
    'Довжина номеру телефону повинна містити 12 символів',
  PHONE_NUMBER_TYPE_MESSAGE: 'Номер повинен містити лише числа',
  ENTER_PHONE_NUMBER_MESSAGE: 'Введіть номер',
  INPUT_LENGTH_MESSAGE: 'Довжина повинна містити не менше 10 символів',
  ENTER_UA_SCHEDULE_MESSAGE: 'Введіть розклад українською',
  ENTER_EN_SCHEDULE_MESSAGE: 'Введіть розклад англійською',
  ENTER_UA_ADDRESS_MESSAGE: 'Введіть адресу українською',
  ENTER_EN_ADDRESS_MESSAGE: 'Введіть адресу англійською',
  IMAGE_FORMAT_MESSAGE:
    'Введіть коректний формат, наприклад: https://example.com/',
  ENTER_LINK_MESSAGE: 'Введіть посилання',
  SELECT_IMAGES_MESSAGE: 'Завантажте зображення для карт'
};

export const emailQuestionsErrorMessages = {
  ANSWER_INPUT_MESSAGE: 'Введіть текст для відповіді'
};

export const materialErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 300 символів`,
  MIN_LENGTH_MESSAGE: `Не менше 2 символів`,
  VALIDATION_ERROR: 'Поле обовязкове',
  PRICE_VALIDATION_ERROR: 'Тільки цифри'
};
export const sizeErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 20 символів`,
  MIN_LENGTH_MESSAGE: `Не менше 1см`,
  MAX_LENGTH_MESSAGE_SIZE: `Не більше 35см`,
  MIN_LENGTH_MESSAGE_SIZE: `Не менше 1 символу`,
  MAX_WEIGHT_MESSAGE_SIZE: `Не більше 5 кг`,
  MIN_WEIGHT_MESSAGE_SIZE: `Не менше 0.1 кг`,
  VALIDATION_ERROR: `Поле обовязкове`,
  PRICE_VALIDATION_ERROR: `Тільки цифри`,
  NOT_UA_INPUT_MESSAGE: `Введіть ім'я розміру українською`,
  NOT_EN_INPUT_MESSAGE: `Введіть ім'я розміру англійською`,
  NO_NUMBER_TYPE_MESSAGE: `Повинно містити число`,
  NO_STRING_TYPE_MESSAGE: `Повинно містити букви`
};

export const colorErrorMessages = {
  MAX_LENGTH_MESSAGE: `Не більше 100 символів`,
  MIN_LENGTH_MESSAGE: `Не менше 2 символів`,
  COLOR_VALIDATION_ERROR: 'Неправильний формат кольору',
  VALIDATION_ERROR: 'Поле обовязкове'
};
export const statsErrorMessages = {
  NO_STATS: 'Статистика для вибраного значення відсутня'
};
export const homePageSlideErrorMessages = {
  SLIDE_VALIDATION_ERROR: 'Мінімум 2 символи',
  SLIDE_ERROR_MESSAGE: 'Поле не може бути порожнім',
  SLIDE_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри'
};

export const constructorErrorMessages = {
  CONSTRUCTOR_VALIDATION_ERROR: 'Мінімум 2 символи',
  CONSTRUCTOR_ERROR_MESSAGE: 'Поле не може бути порожнім',
  CONSTRUCTOR_ERROR_ENGLISH_AND_DIGITS_ONLY: 'Тільки англійські букви і цифри',
  PRICE_VALIDATION_ERROR: 'Тільки цифри',
  PHOTO_NOT_PROVIDED: 'Додайте фото для конструктора'
};
export const paginationInputErrorMessages = {
  MUST_BE_NUMBER: 'Тільки цифри',
  MUST_BE_POSITIVE: 'Тільки більше нуля',
  PAGE_NOT_FOUND: 'Сторінку не знайдено'
};

export const commentErrorMessages = {
  COMMENT_VALIDATION_ERROR: 'Мінімум 2 символи',
  COMMENT_ERROR_MESSAGE: 'Поле не може бути порожнім',
  MAX_LENGTH_MESSAGE: 'Не більше 300 символів'
};
