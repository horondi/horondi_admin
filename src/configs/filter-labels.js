const filterLabels = {
  categories: {
    sortLabels: [
      {
        value: 'sortByNameAsc',
        label: "за ім'ям (у алфавітному порядку)",
        key: 'name',
        type: 'asc'
      },
      {
        value: 'sortByNameDesc',
        label: "за ім'ям (проти алфавітного порядку)",
        key: 'name',
        type: 'desc'
      }
    ]
  },
  orders:{
    users: [
      {
        value: 'sortByNameAsc',
        label: "за ім'ям (у алфавітному порядку)",
        key: 'name',
        type: 'asc'
      },
      {
        value: 'sortByNameDesc',
        label: "за ім'ям (проти алфавітного порядку)",
        key: 'name',
        type: 'desc'
      },
      {
        value: 'sortByEmailAsc',
        label: 'за поштою (у алфавітному порядку)',
        key: 'email',
        type: 'asc'
      },
      {
        value: 'sortByEmailDesc',
        label: 'за поштою (проти алфавітного порядку)',
        key: 'email',
        type: 'desc'
      }
    ],
    sortLabels:[
      {
        value: 'sortByTotalPriceToPayAsc',
        label: "за вартістю товару (по зростанню)",
        key: 'totalPriceToPay.0.value',
        type: 'asc'
      },
      {
        value: 'sortByTotalPriceToPayDesc',
        label: "за вартістю товару (по спаданню)",
        key: 'totalPriceToPay.0.value',
        type: 'desc'
      },
      {
        value: 'sortByCreatedAtAsc',
        label: "від старих до нових",
        key: 'dateOfCreation',
        type: 'asc'
      },
      {
        value: 'sortByCreatedDesc',
        label: "від нових до старих",
        key: 'dateOfCreation',
        type: 'desc'
      },
    ]
  }
};

export default filterLabels;
