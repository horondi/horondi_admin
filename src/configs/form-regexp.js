const formRegExp = {
  patternMaterial: '^[A-Za-z][A-Za-z0-9]*$',
  email:
    '^([\\w-]+(?:\\.[\\w-]+)*)@((?:[\\w-]+\\.)*\\w[\\w-]{0,66})\\.([a-z]{2,6}(?:\\.[a-z]{2})?)$',
  pass: '^(?!.* )(?=.*[0-9])(?=.*[A-Z]).{8,30}$',
  unwrapHtml: /(<([^>]+)>)/gi,
  enAddressRegex: '^[A-Za-z0-9_|,| |./]+$',
  onlyPositiveDigits: /^[1-9]\d*$/,
  editorField: /^<p><br><\/p>|<p><\/p>$/,
  mobileNumber: /^\+380\(\d{2}\)-\d{3}-\d{2}-\d{1,2}$/g,
  userName: /[а-яА-Я]{2,}/g,
  userRoles: /(Користувач|Адмін|Суперадмін)/g,
  userStatuses: /(Активний|Неактивний)/g,
  uaNameCreation: /^[а-яїієґ0-9]+$/i,
  enNameCreation: /^[a-z0-9]+$/i,
  categoryCode: /^[a-z0-9|-]/i
};
export default formRegExp;
