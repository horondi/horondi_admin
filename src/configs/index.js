import { colors } from '@material-ui/core';
import routes from './routes';
import menuCategories from './menu-categories';
import tableHeadRowTitles from './table-head-row-titles';
import detailTitles from './detail-titles';
import buttonTitles from './button-titles';
import statuses from './statuses';
import {
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  patternErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  errorMessages,
  modelErrorMessages,
  headerErrorMessages,
  homePageSlideErrorMessages,
  categoryErrorMessages
} from './error-messages';
import messages from './messages';
import formRegExp from './form-regexp';
import labels from './labels';
import {
  newsPaginationPayload,
  contactsPaginationPayload,
  materialPaginationPayload
} from './pagination';
import titles from './titles';
import { sort } from './sort';

export const config = {
  app: {
    title: 'Horondi Admin Portal',
    serverUrl: 'http://localhost:5000/',
    drawerWidth: 230,
    snackBarDuration: 4000,
    rowsPerPageOptions: [10, 25, 50, 100]
  },
  languages: ['uk', 'en'],
  tokenName: 'HORONDI_AUTH_TOKEN',
  templates: {
    categoryTemplate: {
      available: false,
      code: '',
      images: {
        large: '',
        medium: '',
        small: '',
        thumbnail: ''
      },
      isMain: false,
      name: [],
      subcategories: []
    }
  },
  userRoles: [
    { role: 'user', label: 'Юзер' },
    { role: 'admin', label: 'Адмін' },
    { role: 'superadmin', label: 'Суперадмін' }
  ],
  allowedforRegistrationRoles: ['admin'],
  tabNames: {
    users: ['Користувачі', 'Адміністратори']
  },
  tableSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'medium'
  },
  iconSizes: {
    SMALL_SIZE: 'small',
    DEFAULT_SIZE: 'default'
  },
  theme: {
    DARK_THEME: 'dark',
    LIGHT_THEME: 'light'
  },
  forbiddenRolesFromDeleting: ['superadmin'],
  dialogFormTitles: {
    REGISTER_ADMIN: 'Створити спецкористувача'
  },
  paginationPayload: {
    skip: 0,
    limit: 5,
    countPerPage: 6
  },
  IMG_URL: 'https://horondi.blob.core.windows.net/horondi/images/',
  newsPerPage: 6,
  product: {
    sortBySelectOptions: [
      {
        label: 'популярністю',
        value: 'popularity'
      },
      {
        label: 'від дорогих до дешевих',
        value: 'sortDesc'
      },
      {
        label: 'від дешевих до дорогих',
        value: 'sortAsc'
      },
      {
        label: 'рейтингом',
        value: 'rate'
      }
    ],
    responsive: {
      superLargeDesktop: {
        breakpoint: { max: 4000, min: 3000 },
        items: 1
      },
      desktop: {
        breakpoint: { max: 3000, min: 1024 },
        items: 1
      },
      tablet: {
        breakpoint: { max: 1146, min: 464 },
        items: 1
      },
      mobile: {
        breakpoint: { max: 810, min: 0 },
        items: 1
      }
    }
  },
  popularity: 'popularity',
  rate: 'rate',
  sortAsc: 'sortAsc',
  sortDesc: 'sortDesc',
  submitKey: 'Enter',
  imagePrefix: 'https://horondi.blob.core.windows.net/horondi/images/',
  initialLanguageValues: [
    {
      lang: 'uk',
      value: ''
    },
    {
      lang: 'en',
      value: ''
    }
  ],
  doughnut: {
    colors: [
      colors.green[500],
      colors.red[700],
      colors.amber[600],
      colors.indigo[500]
    ],
    initialValues: {
      names: [],
      counts: [],
      relations: [],
      total: null
    }
  },
  bar: {
    initialValues: {
      labels: [],
      counts: []
    }
  },
  UAH: 'грн',
  buttonTitles,
  detailTitles,
  messages,
  menuCategories,
  routes,
  statuses,
  tableHeadRowTitles,
  formRegExp,
  labels,
  titles,
  loginErrorMessages,
  colorErrorMessages,
  contactErrorMessages,
  patternErrorMessages,
  materialErrorMessages,
  newsErrorMessages,
  categoryErrorMessages,
  errorMessages,
  newsPaginationPayload,
  contactsPaginationPayload,
  materialPaginationPayload,
  modelErrorMessages,
  headerErrorMessages,
  homePageSlideErrorMessages,
  sort
};
export const inputTypes = {
  button: 'button',
  submit: 'submit'
};
