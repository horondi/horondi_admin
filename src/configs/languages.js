const languages = ['ua', 'en'];

export default languages;