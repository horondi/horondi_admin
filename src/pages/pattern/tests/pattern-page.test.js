import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount, to, be } from 'enzyme';
import { Link, BrowserRouter } from 'react-router-dom';
import * as reactRedux from 'react-redux';
import { Button, Typography } from '@material-ui/core';
import mockStore from './mockStore';

import PatternPage from '../pattern-page';
import LoadingBar from '../../../components/loading-bar';
import TableContainerGenerator from '../../../containers/table-container-generator';
import TableContainerRow from '../../../containers/table-container-row';

import { config } from '../../../configs';

const { CREATE_PATTERN_TITLE } = config.buttonTitles;
const pathToPatternAddPage = config.routes.pathToAddPattern;
const tableTitles = config.tableHeadRowTitles.patterns;

configure({ adapter: new Adapter() });

describe('Pattern-page render tests', () => {
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch;
  let wrapper;
  let typography;
  let button;
  let tableContainerGenerator;
  let getState;

  beforeEach(() => {
    // Mock useSelector hook
    spyOnUseSelector = jest.spyOn(reactRedux, 'useSelector');
    spyOnUseSelector.mockImplementation(() => mockStore);

    // Mock useEffect

    // Mock useDispatch hook
    spyOnUseDispatch = jest.spyOn(reactRedux, 'useDispatch');
    // Mock dispatch function returned from useDispatch
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);

    wrapper = shallow(<PatternPage />);
    typography = wrapper.find(Typography);
    button = wrapper.find(Button);
    tableContainerGenerator = wrapper.find(TableContainerGenerator);
  });

  afterEach(() => {
    jest.restoreAllMocks();
    spyOnUseSelector.mockClear();
  });

  test('Should render pattern-page', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  test(`Should render Typography wrapper with "${config.titles.patternTitles.mainPageTitle}" label`, () => {
    expect(wrapper.exists(Typography)).toBe(true);
    expect(typography).toHaveLength(1);
    expect(typography.text()).toBe(config.titles.patternTitles.mainPageTitle);
  });

  test(`Should render button wrapper`, () => {
    expect(wrapper.exists(Button)).toBe(true);
    expect(button).toHaveLength(1);
    expect(button.text()).toBe(CREATE_PATTERN_TITLE);
    expect(button.prop('component')).toEqual(Link);
    expect(button.prop('to')).toBe(pathToPatternAddPage);
  });

  test('Should render TableContainerGenerator', () => {
    expect(wrapper.exists(TableContainerGenerator)).toBe(true);
    expect(tableContainerGenerator).toHaveLength(1);
    expect(wrapper.exists(LoadingBar)).toBe(false);
    expect(tableContainerGenerator.prop('tableTitles')).toBe(tableTitles);
  });

  test('Should render LoadingBar', () => {
    mockStore.loading = true;
    wrapper = shallow(<PatternPage />);
    const loadingBar = wrapper.find(LoadingBar);
    expect(wrapper.exists(LoadingBar)).toBeDefined();
    expect(wrapper.exists(LoadingBar)).toBe(true);
    expect(loadingBar).toHaveLength(1);
    expect(wrapper.exists(TableContainerGenerator)).toBe(false);
    mockStore.loading = false;
  });

  test('useSelector hook should be called', () => {
    getState = reactRedux.useSelector(mockStore);
    expect(getState).toEqual(mockStore);
    expect(spyOnUseSelector).toHaveBeenCalled();
  });
});

describe('useEffect tests', () => {
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch;
  let wrapper;
  let patternPage;
  let tableContainerRow;
  let tableContainerRowFirst;
  let tableContainerRowSecond;
  let tableContainerRowThird;

  beforeEach(() => {
    // Mock useSelector hook
    spyOnUseSelector = jest.spyOn(reactRedux, 'useSelector');
    spyOnUseSelector.mockImplementation(() => mockStore);

    // Mock useDispatch hook
    spyOnUseDispatch = jest.spyOn(reactRedux, 'useDispatch');
    // Mock dispatch function returned from useDispatch
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);

    wrapper = mount(
      <BrowserRouter>
        <PatternPage />
      </BrowserRouter>
    );

    patternPage = wrapper.find(PatternPage);

    tableContainerRow = patternPage.find(TableContainerRow);
    tableContainerRowFirst = patternPage.find({
      id: '6043b87c3e06ad3edcdb7b19'
    });
    tableContainerRowSecond = patternPage.find({
      id: '6043b8c53e06ad3edcdb7b1a'
    });
    tableContainerRowThird = patternPage.find({
      id: '6043b90d3e06ad3edcdb7b1b'
    });
  });

  afterEach(() => {
    wrapper.unmount();
    jest.restoreAllMocks();
    spyOnUseSelector.mockClear();
  });

  test('UseEffect hook shoud work out', () => {
    expect(spyOnUseDispatch).toHaveBeenCalledTimes(2);
  });

  test('Should render TableContainerRow', () => {
    expect(patternPage.exists(TableContainerRow)).toBe(true);
    expect(tableContainerRow).toHaveLength(3);
    expect(tableContainerRowFirst.prop('available')).toBe('Так');
    expect(tableContainerRowSecond.prop('available')).toBe('Так');
    expect(tableContainerRowThird.prop('available')).toBe('Ні');
    expect(tableContainerRowFirst.prop('name')).toBe('Червоний');
    expect(tableContainerRowSecond.prop('name')).toBe('Сірий квадрат');
    expect(tableContainerRowThird.prop('name')).toBe('Голуба стрічка');
    expect(tableContainerRowFirst.prop('image')).toBeTruthy();
    expect(tableContainerRowSecond.prop('image')).toBeTruthy();
    expect(tableContainerRowThird.prop('image')).toBe('');
  });
});
