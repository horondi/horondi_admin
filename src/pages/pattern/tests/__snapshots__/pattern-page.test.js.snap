// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pattern-page render tests Should render pattern-page 1`] = `
<div
  className="makeStyles-container-1"
>
  <div
    className="makeStyles-adminHeader-2"
  >
    <WithStyles(ForwardRef(Typography))
      className="makeStyles-materialTitle-3"
      data-cy="pattern-header"
      variant="h1"
    >
      Інформація про гобелени
    </WithStyles(ForwardRef(Typography))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      component={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      data-cy="add-pattern"
      to="/patterns/add"
      variant="contained"
    >
      Додати гобелен
    </WithStyles(ForwardRef(Button))>
  </div>
  <TableContainerGenerator
    count={7}
    data-cy="patternTable"
    pagination={true}
    tableItems={
      Array [
        <TableContainerRow
          available="Так"
          checkboxChangeHandler={[Function]}
          clickHandler={[Function]}
          deleteHandler={[Function]}
          editHandler={[Function]}
          id=""
          image="https://horondi.blob.core.windows.net/horondi/images/thumbnail_id73cf0klxzl60n_149-min.jpg"
          material="Нитки для шиття"
          name="Червоний"
          showAvatar={true}
          showCheckbox={false}
          showDelete={true}
          showEdit={true}
        />,
        <TableContainerRow
          available="Так"
          checkboxChangeHandler={[Function]}
          clickHandler={[Function]}
          deleteHandler={[Function]}
          editHandler={[Function]}
          id=""
          image="https://horondi.blob.core.windows.net/horondi/images/thumbnail_id73cf0klxzmqds_155-min.jpg"
          material="Нитки для шиття"
          name="Сірий квадрат"
          showAvatar={true}
          showCheckbox={false}
          showDelete={true}
          showEdit={true}
        />,
        <TableContainerRow
          available="Так"
          checkboxChangeHandler={[Function]}
          clickHandler={[Function]}
          deleteHandler={[Function]}
          editHandler={[Function]}
          id=""
          image="https://horondi.blob.core.windows.net/horondi/images/thumbnail_id73cf0klxzoa3x_156-min.jpg"
          material="Нитки для шиття"
          name="Голуба стрічка"
          showAvatar={true}
          showCheckbox={false}
          showDelete={true}
          showEdit={true}
        />,
      ]
    }
    tableTitles={
      Array [
        "Фото",
        "Назва",
        "Код матеріалу",
        "Доступний",
        "Дії",
      ]
    }
  />
</div>
`;
